sklearn/utils/__init__.py

```diff
@@ -268,7 +272,13 @@ def _safe_indexing_row(X, indices):
         else:
```

Comment:
The take path may no longer be more efficient than fancy indexing... And I think these days we can index a sparse matrix with a bool